core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -153,34 +157,19 @@ class NewHadoopRDD[K, V](
           throw new java.util.NoSuchElementException("End of stream")
         }
         havePair = false
-
-        // Update bytes read metric every few records
-        if (recordsSinceMetricsUpdate == HadoopRDD.RECORDS_BETWEEN_BYTES_READ_METRIC_UPDATES
```

Comment:
This was done intentionally to help keep the callback updates out of the `InputMetrics` class and isolate it to Hadoop RDD. This notion of callbacks makes the InputMetrics class more complicated and mutable. Since it's an exposed class we really wanted to keep the interface clean and simple, even if it meant some extra engineering in HadoopRDD.  So could this part of the change be reverted back to how it was before (and you don't change the InputMetrics/TaskMetrics classes?).